py/stream.c

```diff
@@ -75,10 +80,43 @@ mp_uint_t mp_stream_rw(mp_obj_t stream, void *buf_, mp_uint_t size, int *errcode
             return out_sz;
         }
 
+        #if MICROPY_PY_BUILTINS_STR_UNICODE
```

Comment:
This seems like it is fixing a separate issue and so probably deserves a separate pull request.

Reply:
I've removed unicode specific items will add them back in a connected PR that I'll put together next few days.

It would be good to first have this PR merged, so I can add back the unicode tests for stdio which need to removed until unicode support is updated.

Reply:
PR added cover unicode write count here: https://github.com/micropython/micropython/pull/12227